packages/plugin-js-packages/src/lib/runner/audit/unify-type.ts

```diff
@@ -189,11 +191,66 @@ export function yarnv2ToAuditResult(output: string): AuditResult {
     },
   );
 
-  const total = Object.values(yarnv2Audit.metadata.vulnerabilities).reduce(
-    (acc, value) => acc + value,
-    0,
+  return {
+    vulnerabilities,
+    summary: {
+      ...yarnv2Audit.metadata.vulnerabilities,
+      total: getVulnerabilitiesTotal(yarnv2Audit.metadata.vulnerabilities),
+    },
+  };
+}
+
+export function pnpmToAuditResult(output: string): AuditResult {
+  const pnpmResult = JSON.parse(output) as PnpmAuditResultJson;
+
+  const vulnerabilities = Object.values(pnpmResult.advisories).map(
+    ({
+      module_name: name,
+      id,
+      title,
+      url,
+      severity,
+      vulnerable_versions: versionRange,
+      recommendation: fixInformation,
+      findings,
+    }): Vulnerability => {
+      const path = findings[0]?.paths[0];
+
+      return {
+        name,
+        id,
+        title,
+        url,
+        severity,
+        versionRange,
+        directDependency: path == null ? true : pnpmToDirectDependency(path),
+        fixInformation,
+      };
+    },
   );
-  const summary = { ...yarnv2Audit.metadata.vulnerabilities, total };
 
-  return { vulnerabilities, summary };
+  return {
+    vulnerabilities,
+    summary: {
+      ...pnpmResult.metadata.vulnerabilities,
+      total: getVulnerabilitiesTotal(pnpmResult.metadata.vulnerabilities),
+    },
+  };
+}
+
+export function pnpmToDirectDependency(path: string): string | true {
+  // the format is ". > <direct dependency>@<version> > ... > <current dependency>@<version>"
+  const deps = path.split(' ').filter(item => item !== '>' && item !== '.');
+
+  if (deps.length <= 1) {
```

Comment:
```suggestion
  if (deps.length === 0) {
```

Pretty sure the `if`-statement doesn't change the result, though.

Reply:
It should.
If the original is `. > cypress@13.7.0` then it is a direct dependency. Then the `deps.length === 1` and it should return `true`.
If the original is `. > cypress@13.7.0 > @cypress/request@3.0.0` then the `deps.length === 2` and it should return `cypress` as the name of the first dependency after `.`.

Reply:
Ah, ignore my comment, I misread the code :grin: My comment would've made sense if the condition was `deps.length < 1`, but I now realize it's `deps.length <= 1` :facepalm: